sdk/how-to-guides/ports.md

```diff
@@ -0,0 +1,144 @@
+> See first: [Juju `open-port`](https://juju.is/docs/juju/hook-tool#open-port)
+
+**Contents**:
+
+1. [Open a port based on the charm config](#heading--open-a-port-to-your-charm)
+1. [Test the port is opened](#heading--test-the-port-is-opened)
+
+<a href="#heading--heading--open-a-port-to-your-charm"><h2 id="heading--heading--open-a-port-to-your-charm">Open a port based on the charm config</h2></a>
+
+To make a service permanently reachable under a stable URL on the cluster the charm needs to open (expose) a port. This ensures that the charm will be consistently accessible even if the pod gets recycled and the IP address changes.
+
+The port that the charm's service should be reachable on is typically a config option; for example, by including a section like this in `charmcraft.yaml`:
```

Comment:
From here onwards, what follows is a version of the how-to guide for how to add a config to a charm adapted for defining a port. I'd make this more explicit and then give the details as details. E.g.,

The port that the charm's service should be reachable on is typically defined as a config option. As such, all the usual procedure for adding a config option to a charm applies:

1. In your `charmcraft.yaml` define a config that sets the port to be opened/exposed. For example:
...
2. In your `src/charm.py` observe the `config_changed` event and define a handler. For example:
...

> Further examples: ...
> 
> See more: [How to add a config option to a charm](/t/12244) 

(Note that the see more link is not to the charmcraft / ops constructs but rather to the generic how-to guide. I think that is in fact the correct thing to link to here because it'll help people not just connect to the specific reference bits for this but also the general add config story, including testing. Speaking of which: My guess is the same general comments will apply to the test section below as well.)